test_regress/t/t_vpi_var.cpp

```diff
@@ -644,16 +744,21 @@ int mon_check() {
     printf("-mon_check()\n");
 #endif
 
-    if (int status = _mon_check_mcd()) return status;
-    if (int status = _mon_check_callbacks()) return status;
-    if (int status = _mon_check_value_callbacks()) return status;
+    if (TestSimulator::is_verilator()) {
+        if (int status = _mon_check_mcd()) return status;
+        if (int status = _mon_check_callbacks()) return status;
+        if (int status = _mon_check_value_callbacks()) return status;
+    }
```

Comment:
Why are you not also testing the other simulators for these?  It's important we be able to compare results with others - can you get it passing without this if, and instead if you have to, push the "if" down to the smallest possible code segment.  Thanks

Reply:
I agree, but this does not work on master either (at least, with Icarus). I will try to limit this if as much as possible, and to make it work with ModelSim also, it should reduce the number of things which are not compared.

Reply:
OK, the test works with ModelSIm (apart from a few change to make, mainly shell creating and giving more time for the test). 

I will skip all elements Icarus fails on to make it work with Icarus. I can't check with other simulators. I push the result as soon as it is available.